api/clients/disperser_client.go

```diff
@@ -273,18 +284,25 @@ func (c *disperserClient) GetBlobStatus(ctx context.Context, requestID []byte) (
 	return c.client.GetBlobStatus(ctxTimeout, request)
 }
 
-func (c *disperserClient) RetrieveBlob(ctx context.Context, batchHeaderHash []byte, blobIndex uint32) ([]byte, error) {
+func (c *DisperserClient) RetrieveBlob(ctx context.Context, batchHeaderHash []byte, blobIndex uint32) ([]byte, error) {
 	err := c.initOnceGrpcConnection()
 	if err != nil {
 		return nil, api.NewErrorInternal(err.Error())
 	}
 
 	ctxTimeout, cancel := context.WithTimeout(ctx, time.Second*60)
 	defer cancel()
-	reply, err := c.client.RetrieveBlob(ctxTimeout, &disperser_rpc.RetrieveBlobRequest{
-		BatchHeaderHash: batchHeaderHash,
-		BlobIndex:       blobIndex,
-	})
+	if c.config.MaxRetrieveBlobSizeBytes == 0 {
```

Comment:
should we have a flag for this? this looks adhoc

Reply:
commented in the wrong place, make c.config.MaxRetrieveBlobSizeBytes set in the constructor with the flag, so no if statement here

Reply:
I mean MaxRetrieveBlobSizeBytes is always 0. since there is no flag to set it

Reply:
It's not. constructor takes a config, which could have set MaxRetrieveBlobSizeBytes
```
func NewDisperserClient(config *Config, signer core.BlobRequestSigner) *DisperserClient {
```

And as I said above, I prefer not to set in constructor, because one can still build a DisperserClient without using the constructor. This is often done in tests for eg, like
```
client := &DisperserClient{
   Conn: ...
   Foo: ...
   
```

Reply:
Ahhhh scratch that... fields are private, so constructor actually has to be used. Ok, you're right, will update

Reply:
[04499a5](https://github.com/Layr-Labs/eigenda/pull/849/commits/04499a54c72071151d1bee67b1ba8e68d98328c6)